File: src/modules/workflow/application/workflow_application/dto/workflow_application_dto.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID


@dataclass(frozen=True, slots=True)
class WorkflowApplicationDTO:
    """DTO for workflow application data."""

    id: UUID

    created_at: datetime
    updated_at: datetime

    created_by: UUID | None
    updated_by: UUID | None

    kind: str
    status: str

    title: str
    description: str | None

    icon: str
    icon_background: str

    active_workflow_definition_id: UUID | None


__all__ = ["WorkflowApplicationDTO"]

File: frontends/apps/console/src/modules/workflow/applications/api/workflow-application.dto.ts
/**
 * DTO — то, что приходит с API.
 */
export interface WorkflowApplicationListItemDto {
  id: string;
  created_at: string;
  kind: string;
  status: string;
  title: string;
  description: string | null;
  icon: string;
  icon_background: string;
}

export interface WorkflowApplicationListResponseDto {
  items: WorkflowApplicationListItemDto[];
  next_cursor: string | null;
}

export interface WorkflowApplicationResponseDto {
  id: string;
  created_at: string;
  updated_at: string;
  created_by: string | null;
  updated_by: string | null;
  kind: string;
  status: string;
  title: string;
  description: string | null;
  icon: string;
  icon_background: string;
  active_workflow_definition_id: string | null;
}

export interface CreateWorkflowApplicationRequestDto {
  title: string;
  description: string | null;
  icon: string;
  icon_background: string;
}
